workitem/event/event_repository.go

```diff
@@ -0,0 +1,119 @@
+package event
+
+import (
+	"context"
+
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+const APIStringTypeEvents = "events"
+
+// WorkItemEventRepository encapsulates retrieval of work item events
+type WorkItemEventRepository interface {
+	//repository.Exister
+	List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error)
+}
+
+// NewWorkItemEventRepository creates a work item event repository based on gorm
+func NewWorkItemEventRepository(db *gorm.DB) *GormWorkItemEventRepository {
+	return &GormWorkItemEventRepository{
+		db:               db,
+		workItemRepo:     workitem.NewWorkItemRepository(db),
+		wiRevisionRepo:   workitem.NewRevisionRepository(db),
+		workItemTypeRepo: workitem.NewWorkItemTypeRepository(db),
+		identityRepo:     account.NewIdentityRepository(db),
+	}
+}
+
+type GormWorkItemEventRepository struct {
+	db               *gorm.DB
+	workItemRepo     *workitem.GormWorkItemRepository
+	wiRevisionRepo   *workitem.GormRevisionRepository
+	workItemTypeRepo *workitem.GormWorkItemTypeRepository
+	identityRepo     *account.GormIdentityRepository
+}
+
+func (r *GormWorkItemEventRepository) List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error) {
+	revisionList, err := r.wiRevisionRepo.List(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	if revisionList == nil {
+		return []WorkItemEvent{}, nil
+	}
+	wi, err := r.workItemRepo.LoadByID(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	wiType, err := r.workItemTypeRepo.LoadByID(ctx, wi.Type)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+
+	eventList := []WorkItemEvent{}
+	for k := 1; k < len(revisionList); k++ {
```

Comment:
Why not `for k, revision := range revisionList` ?

Reply:
The index need to start at 1 as per the logic.